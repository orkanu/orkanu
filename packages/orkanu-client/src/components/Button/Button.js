import React from 'react'
import styled from '@emotion/styled'

const colors = {
    primary: {
        black: '#000000',
        orange: '#ff9900',
        blue: '#146eb4',
    },
    neutral: {
        white: '#ffffff',
    },
}

const blockSize = s => s * 8

const buttonStyles = {
    primary: `
    min-width: ${blockSize(18)}px;
    background-color: ${colors.primary.black};
    color: ${colors.neutral.orange};

    :hover {
      background-color: #0088F1;
    }

    :active {
      background-color: #0061AB;
    }
  `,
    secondary: `
    min-width: ${blockSize(18)}px;
    border: 1px solid ${colors.primary.black};
    background-color: transparent;
    color: ${colors.primary.black};

    :hover {
      color: #0088F1;
      border-color: #0088F1;
    }

    :active {
      background-color: #0061AB;
      color: ${colors.neutral.white};
    }
  `,
    tertiary: `
    background-color: transparent;
    text-decoration: underline;
    color: ${colors.primary.black};

    :hover {
      color: #0088F1;
    }

    :active {
      color: #0061AB;
    }
  `,
}
const getStylesForButton = ({ uiType }) => buttonStyles[uiType]

const microStyles = `
  min-width: auto;
  white-space: nowrap;
  margin-left: ${blockSize(1)}px;
  font-family: 'Source Sans Pro', Helvetica, Sans-Serif;
  font-weight: 300;
  padding: 2px ${blockSize(0.5)}px;
`

const StyledButton = styled('button')`
    font-family: 'Source Sans Pro', Helvetica, Sans-Serif;
    font-weight: 600;
    border: 0;
    padding: ${blockSize(1.25)}px ${blockSize(2)}px;
    border-radius: ${blockSize(0.5)}px;
    text-decoration: none;
    text-align: center;
    ${getStylesForButton};
    ${props => (props.uiSize === 'micro' ? microStyles : ``)};
    :disabled {
        opacity: 0.5;
        pointer-events: none;
    }
`

const Button = ({
    dataTestId,
    children,
    parentId,
    uiType = 'primary',
    uiSize,
    disabled,
    onClick,
    tag = 'button',
    href,
}) => {
    return (
        <StyledButton
            data-test-id={dataTestId}
            id={parentId && `${parentId}-button`}
            uiSize={uiSize}
            uiType={uiType}
            disabled={disabled}
            onClick={onClick}
            as={tag}
            href={href}
        >
            {children}
        </StyledButton>
    )
}

export { Button }
