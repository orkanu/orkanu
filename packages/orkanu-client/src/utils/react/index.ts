import React from 'react'

export const renderChildren = (children: any, id?: string) => {
    return React.Children.map(children, (child, key) => {
        return React.cloneElement(child, {
            key,
            index: key,
            parentId: id,
        })
    })
}
