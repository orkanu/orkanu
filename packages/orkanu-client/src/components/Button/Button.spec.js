import React from 'react';
import { render, cleanup, configure } from '@testing-library/react';
import { Button } from './Button';

configure({ testIdAttribute: 'data-test-id' });

describe('Button', () => {
  const renderComponent = props => {
    const utils = render(<Button {...props}></Button>);
    const testee = utils.getByTestId('test');
    return { ...utils, testee };
  };

  let testeeProps;
  beforeEach(() => {
    testeeProps = { dataTestId: 'test' };
  });
  afterEach(() => {
    testeeProps = null;
    cleanup();
  });
  test('renders correctly', () => {
    const { testee } = renderComponent(testeeProps);
    expect(testee).toMatchSnapshot();
  });
});
