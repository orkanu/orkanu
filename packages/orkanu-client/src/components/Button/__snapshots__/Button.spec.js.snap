// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
.emotion-0 {
  font-family: "Source Sans Pro",Helvetica,Sans-Serif;
  font-weight: 600;
  font-size: 16px;
  line-height: 20px;
  border: 0;
  padding: 10px 16px;
  border-radius: 4px;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-align: center;
  min-width: 144px;
  background-color: #0072C9;
  color: #FFFFFF;
}

.emotion-0:hover {
  background-color: #0088F1;
}

.emotion-0:active {
  background-color: #0061AB;
}

.emotion-0:disabled {
  opacity: 0.5;
  pointer-events: none;
}

<button
  class="emotion-0 emotion-1"
  data-test-id="test"
/>
`;
